nativelib/src/main/resources/gc/immix/headers/LineHeader.h

```diff
@@ -0,0 +1,58 @@
+#ifndef IMMIX_LINEHEADER_H
+#define IMMIX_LINEHEADER_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#define FIRST_OBJECT_OFFSET_MASK (uint8_t)0xFC
+
+typedef struct {
+    int16_t next;
+    uint16_t size;
+} FreeLineHeader;
+
+typedef enum {
+    line_empty = 0x0,
+    line_marked = 0x1,
+    line_contains_object = 0x2,
```

Comment:
It should rather be renamed to `line_contains_object_header`. It an object spans multiple lines, only the line that contains the object header will have this flag set.